hazelcast/test/src/compact_test.cpp

```diff
@@ -734,6 +1096,37 @@ TEST_F(CompactSerializationTest, test_rabin_fingerprint_consistent_with_server)
     ASSERT_EQ(814479248787788739L, schema.schema_id());
 }
 
+TEST_F(CompactSerializationTest, test_read_when_field_does_not_exist)
+{
+    using hazelcast::client::compact::test::wrong_field_name_read_obj;
+
+    serialization_config config;
+    SerializationService ss(config);
+
+    wrong_field_name_read_obj obj;
+
+    auto data = ss.to_data(obj);
+    ASSERT_THROW(
+        ss.to_object<wrong_field_name_read_obj>(data),
+        exception::hazelcast_serialization
+    );
+}
+
+TEST_F(CompactSerializationTest, test_read_with_type_mismatch)
```

Comment:
Should we add test case for two types which has same size. For example, while writing int32 and reading float value, will it give an exception too?

Reply:
Fixed.